tests/unit/Imports/ProductImportTest.php

```diff
@@ -155,6 +157,194 @@ function (MockInterface $module) {
     }
 
+    public function dataProviderTestGetStockProperties()
+    {
+        $tests = [];
+
+        $tests['simple or configurable assign product in stock, limited, has positive orderable value'] = [
+            [
+                'type' => 'simple',
+                'product_stock' => [
+                    'in_stock' => true,
+                    'unlimited' => false,
+                    'value' => 5,
+                ],
+                'orderable_stock_value' => 10,
+            ],
+            [
+                'in_stock' => true,
+                'manage_stock' => true,
+                'quantity' => 10,
+            ],
+        ];
+
+        $tests['simple or configurable assign product in stock, limited, has negative orderable value'] = [
+            [
+                'type' => 'simple',
+                'product_stock' => [
+                    'in_stock' => true,
+                    'unlimited' => false,
+                    'value' => 5,
+                ],
+                'orderable_stock_value' => -10,
+            ],
+            [
+                'in_stock' => false,
+                'manage_stock' => true,
+                'quantity' => 0,
+            ],
+        ];
+
+        $tests['simple or configurable assign product in stock, limited, has no orderable value'] = [
+            [
+                'type' => 'configurable_assign',
+                'product_stock' => [
+                    'in_stock' => true,
+                    'unlimited' => false,
+                    'value' => 5,
+                ],
+                'orderable_stock_value' => null,
+            ],
+            [
+                'in_stock' => true,
+                'manage_stock' => true,
+                'quantity' => 5,
+            ],
+        ];
+
+        $tests['simple or configurable assign product out stock, limited, has orderable value'] = [
+            [
+                'type' => 'simple',
+                'product_stock' => [
+                    'in_stock' => false,
+                    'unlimited' => false,
+                    'value' => 5, // Ignored cause in_stock is false
+                ],
+                'orderable_stock_value' => 10, // Ignored cause in_stock is false
+            ],
+            [
+                'in_stock' => false,
+                'manage_stock' => true,
+                'quantity' => 0,
+            ],
+        ];
+
+        $tests['simple or configurable assign product out stock, unlimited, has orderable value'] = [
+            [
+                'type' => 'simple',
+                'product_stock' => [
+                    'in_stock' => false,
+                    'unlimited' => true, // Ignored cause in_stock is false
+                    'value' => 5,
+                ],
+                'orderable_stock_value' => 10, // Ignored cause in_stock is false
+            ],
+            [
+                'in_stock' => false,
+                'manage_stock' => true,
```

Comment:
```suggestion
                'manage_stock' => false,
```
it should be false here, because unlimited